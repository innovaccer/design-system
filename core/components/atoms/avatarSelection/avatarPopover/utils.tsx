import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
