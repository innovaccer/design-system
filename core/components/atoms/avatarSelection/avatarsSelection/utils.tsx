import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
