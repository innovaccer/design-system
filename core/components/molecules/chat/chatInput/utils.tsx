import { MessageType } from './ChatInput';

export const positionMentionPopup = (
  textareaRef: React.RefObject<HTMLDivElement>,
  setMentionPosition: React.Dispatch<React.SetStateAction<{ top: number; left: number }>>
) => {
  if (!textareaRef.current) return;

  const selection = window.getSelection();
  if (!selection || !selection.rangeCount) return;

  const range = selection.getRangeAt(0);
  const rect = range.getBoundingClientRect();
  const editorRect = textareaRef.current.getBoundingClientRect();

  // Calculate position based on the cursor position
  let top = rect.bottom + window.scrollY;
  let left = rect.left + window.scrollX;

  // Adjust position if it would go off-screen
  if (top + 150 > window.innerHeight) {
    top = rect.top + window.scrollY - 150;
  }
  if (left + 200 > window.innerWidth) {
    left = window.innerWidth - 200;
  }

  // Adjust position relative to the editor
  top -= editorRect.top + window.scrollY - 12;
  left -= editorRect.left + window.scrollX - 8;

  setMentionPosition({ top, left });
};

export const extractMessageData = (textareaRef: React.RefObject<HTMLDivElement>): MessageType[] => {
  const elements = textareaRef.current?.childNodes;

  return Array.from(elements || []).map((item: any) => {
    if (item?.dataset?.type === 'mention') {
      const mentionValue = JSON.parse(item.dataset.content);
      return { type: 'mention', content: mentionValue?.data, nodeName: item.nodeName };
    }

    // handles new line by pressing enter
    else if (item.innerText === '\n' && item.nodeName === 'DIV') {
      return { type: 'newline', content: '\n', nodeName: item.nodeName };
    }

    // handles line break by pressing shift + enter
    else if (item.nodeName === 'BR') {
      return { type: 'linebreak', content: '<br>', nodeName: item.nodeName };
    }

    return { type: 'text', content: item.innerText || item.textContent, nodeName: item.nodeName };
  });
};
