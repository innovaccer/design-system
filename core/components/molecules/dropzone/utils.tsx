const isIe = (userAgent: string) => {
  return userAgent.indexOf('MSIE') !== -1 || userAgent.indexOf('Trident/') !== -1;
};

const isEdge = (userAgent: string) => {
  return userAgent.indexOf('Edge/') !== -1;
};

export const accepts = (file: File, acceptedFiles?: string | string[]) => {
  if (file && acceptedFiles) {
    const acceptedFilesArray = Array.isArray(acceptedFiles) ? acceptedFiles : acceptedFiles.split(',');
    const fileName = file.name || '';
    const mimeType = (file.type || '').toLowerCase();
    const baseMimeType = mimeType.replace(/\/.*$/, '');

    if (!mimeType) {
      return true;
    }

    return acceptedFilesArray.some((type) => {
      const validType = type.trim().toLowerCase();
      if (validType.charAt(0) === '.') {
        return fileName.toLowerCase().endsWith(validType);
      }
      if (validType.endsWith('/*')) {
        return baseMimeType === validType.replace(/\/.*$/, '');
      }
      return mimeType === validType;
    });
  }
  return true;
};

export const isPropagationStopped = (event: any) => {
  if (typeof event.isPropagationStopped === 'function') {
    return event.isPropagationStopped();
  }
  if (typeof event.cancelBubble !== 'undefined') {
    return event.cancelBubble;
  }
  return false;
};

export const isEvtWithFiles = (event: any) => {
  if (!event.dataTransfer) {
    return !!event.target && !!event.target.files;
  }

  return Array.prototype.some.call(
    event.dataTransfer.types,
    (type) => type === 'Files' || type === 'application/x-moz-file'
  );
};

export const isKindFile = (item: null | Record<string, unknown>) => {
  return typeof item === 'object' && item !== null && item.kind === 'file';
};

export const onDocumentDragOver = (event: DragEvent) => {
  event.preventDefault();
};

export const isIeOrEdge = (userAgent = window.navigator.userAgent) => {
  return isIe(userAgent) || isEdge(userAgent);
};

type functype = (event: Event, ...args: unknown[]) => void;

export const composeEventHandlers = (...fns: functype[]) => {
  return (event: Event, ...args: unknown[]) =>
    fns.some((fn) => {
      if (!isPropagationStopped(event) && fn) {
        fn(event, ...args);
      }
      return isPropagationStopped(event);
    });
};

export const reducer = (state: any, action: any) => {
  switch (action.type) {
    case 'focus':
      return {
        ...state,
        isFocused: true,
      };
    case 'blur':
      return {
        ...state,
        isFocused: false,
      };
    case 'openDialog':
      return {
        ...state,
        isFileDialogActive: true,
      };
    case 'closeDialog':
      return {
        ...state,
        isFileDialogActive: false,
      };
    case 'setDraggedFiles':
      /* eslint no-case-declarations: 0 */
      const { isDragActive, draggedFiles } = action;
      return {
        ...state,
        draggedFiles,
        isDragActive,
      };
    case 'setFiles':
      return {
        ...state,
        acceptedFiles: action.acceptedFiles,
        fileRejections: action.fileRejections,
      };
    case 'reset':
      return {
        ...state,
        isFileDialogActive: false,
        isDragActive: false,
        draggedFiles: [],
        acceptedFiles: [],
        fileRejections: [],
      };
    default:
      return state;
  }
};
