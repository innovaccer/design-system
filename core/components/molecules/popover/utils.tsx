import { Placement } from '@floating-ui/react';
import { PositionType as Position } from '@/common.type';

export const mapPositionToPlacement = (position: Position): Placement => {
  const positionMap: Record<Position, Placement> = {
    auto: 'bottom',
    'auto-start': 'bottom-start',
    'auto-end': 'bottom-end',
    'top-start': 'top-start',
    top: 'top',
    'top-end': 'top-end',
    'right-start': 'right-start',
    right: 'right',
    'right-end': 'right-end',
    'bottom-end': 'bottom-end',
    bottom: 'bottom',
    'bottom-start': 'bottom-start',
    'left-end': 'left-end',
    left: 'left',
    'left-start': 'left-start',
  };

  return positionMap[position] || 'bottom';
};
