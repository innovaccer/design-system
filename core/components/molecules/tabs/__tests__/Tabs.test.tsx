import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import { Tabs, Tab } from '@/index';
import { TabsProps as Props, TabConfig } from '@/index.type';
import { TTabSize } from '@/common.type';

const activeIndex = 0;
const FunctionValue = jest.fn();
const BooleanValue = [true, false];
const tabSize: TTabSize[] = ['regular', 'small'];
const tabs = [
  {
    count: 10,
    label: 'Tab 1',
    isDismissible: true,
    onDismiss: FunctionValue,
  },
  {
    icon: 'call_received',
    label: 'Tab 2',
  },
  {
    label: 'Tab 3',
    icon: 'call_received',
    count: 10,
    disabled: true,
  },
];

describe('Tabs component', () => {
  const Mapper: Record<string, any> = {
    activeIndex: valueHelper(activeIndex, { required: true }),
    onTabChange: valueHelper(FunctionValue, { required: true }),
    withSeparator: valueHelper(BooleanValue, { required: true, iterate: true }),
    size: valueHelper(tabSize, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { asFragment } = render(<Tabs {...attr} tabs={tabs} />);
      expect(asFragment()).toMatchSnapshot();
    });
  };

  testHelper(Mapper, testFunc);
});

describe('Tabs component', () => {
  it('renders tab', () => {
    const activeTab = 0;
    const activeTabContent = tabs[activeTab].label;

    const { getAllByTestId } = render(<Tabs activeIndex={activeTab} onTabChange={FunctionValue} tabs={tabs} />);

    expect(getAllByTestId('DesignSystem-Tabs--Text')[activeTab].textContent).toMatch(activeTabContent);
  });

  it('renders tab with count', () => {
    const countValue = '10';

    const { getAllByTestId } = render(<Tabs tabs={tabs} />);

    expect(getAllByTestId('DesignSystem-Tabs--Pills')[0].textContent).toMatch(countValue);
  });

  it('renders tab with icon', () => {
    const iconName = 'call_received';

    const { getByTestId } = render(<Tabs tabs={tabs} />);

    expect(getByTestId('DesignSystem-Tabs--Icon').textContent).toMatch(iconName);
  });

  it('renders count if tab object has both icon and count', () => {
    const count = '10';

    const { getAllByTestId } = render(<Tabs tabs={tabs} />);

    expect(getAllByTestId('DesignSystem-Tabs--Pills')).toHaveLength(2);
    expect(getAllByTestId('DesignSystem-Tabs--Pills')[1].textContent).toMatch(count);
  });

  it('renders active tab and children', () => {
    const activeTab = 1;

    const { getByTestId, getAllByTestId } = render(
      <Tabs activeIndex={activeTab} onTabChange={FunctionValue} tabs={tabs} />
    );

    expect(getAllByTestId('DesignSystem-Tabs--Tab')[activeTab]).toHaveClass('Tab--active');
    expect(getByTestId('DesignSystem-Tabs--Icon')).toHaveClass('Icon--info');
  });

  it('renders default tab as 0', () => {
    const defaultTab = 0;

    const { getAllByTestId } = render(<Tabs onTabChange={FunctionValue} tabs={tabs} />);

    expect(getAllByTestId('DesignSystem-Tabs--Tab')[defaultTab]).toHaveClass('Tab--active');
  });

  it('default tab is 0 if activeTab > total tabs length', () => {
    const activeTab = 3;
    const defaultActiveTab = 0;

    const { getAllByTestId } = render(<Tabs onTabChange={FunctionValue} tabs={tabs} activeIndex={activeTab} />);

    expect(getAllByTestId('DesignSystem-Tabs--Tab')[defaultActiveTab]).toHaveClass('Tab--active');
  });

  it('renders disabled tab', () => {
    const activeTab = 2;
    const disabledTab = 2;

    const { getAllByTestId } = render(<Tabs onTabChange={FunctionValue} tabs={tabs} activeIndex={activeTab} />);

    expect(getAllByTestId('DesignSystem-Tabs--Tab')[activeTab]).not.toHaveClass('Tab--active');
    expect(getAllByTestId('DesignSystem-Tabs--Tab')[disabledTab]).toHaveClass('Tab--disabled');
    expect(getAllByTestId('DesignSystem-Tabs--Pills')[1]).toHaveClass('Tab-pills--disabled');
  });

  it('renders dismissible tab component in active state', () => {
    const activeTab = 0;
    const { getByTestId } = render(<Tabs activeIndex={activeTab} onTabChange={FunctionValue} tabs={tabs} />);
    expect(getByTestId('DesignSystem-DismissibleTabs--Icon')).toHaveClass('DismissibleTab-icon--selected');
  });

  it('renders dismissible tab component', () => {
    const activeTab = 1;
    const { getByTestId } = render(<Tabs activeIndex={activeTab} onTabChange={FunctionValue} tabs={tabs} />);
    expect(getByTestId('DesignSystem-DismissibleTabs--Icon')).toHaveClass('DismissibleTab-icon--default');
  });
});

describe('TabsWrapper component with prop: onTabChange', () => {
  afterEach(() => jest.clearAllMocks());
  const getTestTab = (index: number, testTabs: TabConfig[]) => (
    <Tabs activeIndex={index} tabs={testTabs} onTabChange={FunctionValue} />
  );

  it('TabsWrapper component with onTabChange callback', () => {
    const activeTab = 0;
    const clickedTab = 1;

    const { getAllByTestId, rerender } = render(getTestTab(activeTab, tabs));

    const tab = getAllByTestId('DesignSystem-Tabs--Tab')[clickedTab];
    fireEvent.click(tab);

    expect(FunctionValue).toHaveBeenCalledWith(clickedTab);
    rerender(getTestTab(clickedTab, tabs));
    expect(getAllByTestId('DesignSystem-Tabs--Tab')[clickedTab]).toHaveClass('Tab--active');
  });

  it('TabsWrapper component with onTabChange callback on enter press', () => {
    const activeTab = 0;
    const KeyedTab = 1;

    const { getAllByTestId, rerender } = render(getTestTab(activeTab, tabs));

    const tab = getAllByTestId('DesignSystem-Tabs--Tab');
    fireEvent.keyDown(tab[KeyedTab], { key: 'Enter', keyCode: 13 });

    expect(FunctionValue).toHaveBeenCalledWith(KeyedTab);
    rerender(getTestTab(KeyedTab, tabs));
    expect(tab[KeyedTab]).toHaveClass('Tab--active');
  });

  it('TabsWrapper component with onTabChange callback not called on any other key', () => {
    const activeTab = 0;
    const KeyedTab = 1;

    const { getAllByTestId, rerender } = render(getTestTab(activeTab, tabs));
    const tab = getAllByTestId('DesignSystem-Tabs--Tab');
    fireEvent.keyDown(tab[KeyedTab], { key: 'Space', keyCode: 32 });

    expect(FunctionValue).not.toHaveBeenCalledWith(KeyedTab);
    rerender(getTestTab(activeTab, tabs));
    expect(tab[KeyedTab]).not.toHaveClass('Tab--active');
  });

  it('TabsWrapper component changes tab right when ArrowRight is pressed', () => {
    const activeTab = 0;

    const { getAllByTestId } = render(getTestTab(activeTab, tabs));
    const tab = getAllByTestId('DesignSystem-Tabs--Tab');
    const firstLabel = tab[activeTab];
    const secondLabel = tab[1];

    fireEvent.focus(firstLabel);
    fireEvent.keyDown(firstLabel, { key: 'ArrowRight', keyCode: 39 });

    expect(document.activeElement).toContainElement(secondLabel);
  });

  it('TabsWrapper component changes tab left when ArrowLeft is pressed', () => {
    const activeTab = 1;

    const { getAllByTestId } = render(getTestTab(activeTab, tabs));
    const tab = getAllByTestId('DesignSystem-Tabs--Tab');
    const firstLabel = tab[0];
    const secondLabel = tab[1];

    fireEvent.focus(secondLabel);
    fireEvent.keyDown(secondLabel, { key: 'ArrowLeft', keyCode: 37 });

    expect(document.activeElement).toContainElement(firstLabel);
  });

  it('TabsWrapper component has only tabindex 0 for all tabs', () => {
    const activeTab = 0;
    const keyedTab = 1;

    const { getAllByTestId, rerender } = render(getTestTab(activeTab, tabs));
    const tab = getAllByTestId('DesignSystem-Tabs--Tab');
    const firstLabel = tab[activeTab];
    const secondLabel = tab[keyedTab];

    expect(firstLabel.tabIndex).toBe(0);
    expect(secondLabel.tabIndex).toBe(0);

    fireEvent.focus(secondLabel);
    fireEvent.keyDown(secondLabel, { key: 'ArrowLeft', keyCode: 37 });

    rerender(getTestTab(keyedTab, tabs));

    expect(firstLabel.tabIndex).toBe(0);
    expect(secondLabel.tabIndex).toBe(0);
    expect(document.activeElement).toContainElement(firstLabel);
  });

  it('renders tab component with isDismissible prop:true', () => {
    const { getAllByTestId } = render(<Tabs tabs={tabs} />);
    expect(getAllByTestId('DesignSystem-DismissibleTabs--Icon')[0]).toBeInTheDocument();
  });

  it('renders tab component with isDismissible prop:true and calls onDismiss', () => {
    const { getAllByTestId } = render(<Tabs tabs={tabs} />);
    const tab = getAllByTestId('DesignSystem-DismissibleTabs--Icon')[0];
    fireEvent.click(tab);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('render tab component with custom class', () => {
    const { getAllByTestId } = render(
      <Tabs activeIndex={0}>
        <Tab label="Tab(Recommended)" isDismissible={true} className="custom-class" onDismiss={FunctionValue}></Tab>
      </Tabs>
    );
    expect(getAllByTestId('DesignSystem-Tabs--Content')[0]).toHaveClass('custom-class');
  });

  it('renders tab component with remaining tab after onDismiss call ', () => {
    const tabList = [
      {
        count: 10,
        label: 'Tab 1',
        icon: 'call_received',
      },
      {
        count: 10,
        label: 'Tab 2',
        isDismissible: true,
        onDismiss: FunctionValue,
      },
    ];

    const { getAllByTestId } = render(<Tabs tabs={tabList} />);
    const activeTab = getAllByTestId('DesignSystem-Tabs--Tab')[1];
    fireEvent.click(activeTab);

    const clearIcon = getAllByTestId('DesignSystem-DismissibleTabs--Icon')[0];
    fireEvent.click(clearIcon);

    expect(FunctionValue).toHaveBeenCalled();
    expect(getAllByTestId('DesignSystem-Tabs--Tab')[0]).toBeInTheDocument();
  });
});

describe('Tabs Wrapper component header class', () => {
  it('render tab component header with custom class', () => {
    const { getByTestId } = render(
      <Tabs activeIndex={0} headerClassName="header-class">
        <Tab label="Tab(Recommended)" isDismissible={true} onDismiss={FunctionValue}></Tab>
      </Tabs>
    );
    expect(getByTestId('DesignSystem-Tabs--Header')).toHaveClass('header-class');
  });
});

describe('Tabs component with prop:size', () => {
  tabSize.forEach((size) => {
    it(`should have the Tab--${size} class when size=${size}`, () => {
      const { getByTestId } = render(
        <Tabs activeIndex={0} size={size}>
          <Tab label="Tab(Recommended)" isDismissible={true} onDismiss={FunctionValue}></Tab>
        </Tabs>
      );
      expect(getByTestId('DesignSystem-Tabs--TextWrapper')).toHaveClass(`Tab--${size}`);
    });
  });
});

describe('Tabs Wrapper component custom max-width', () => {
  it('render tab component header with custom max-width', () => {
    const { getByTestId } = render(
      <Tabs activeIndex={0} maxWidth={100}>
        <Tab label="Tab(Recommended)" isDismissible={true} onDismiss={FunctionValue}></Tab>
      </Tabs>
    );
    expect(getByTestId('DesignSystem-Tabs--TextWrapper')).toHaveStyle('max-width: 100px;');
  });
});
