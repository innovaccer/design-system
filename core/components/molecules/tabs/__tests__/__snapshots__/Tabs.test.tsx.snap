// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component 
  activeIndex: 0, onTabChange: "[Function]", withSeparator: false, size: "regular"
 1`] = `
<DocumentFragment>
  <div
    class="TabsWrapper"
    data-test="DesignSystem-Tabs"
  >
    <div
      class="TabsWrapper-header"
      data-test="DesignSystem-Tabs--Header"
    >
      <div
        class="Tab Tab--active Tab-selected align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="0"
      >
        <span
          class="Tab--regular Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <span
            class="Pills Badge--primary Tab-pills"
            data-test="DesignSystem-Tabs--Pills"
          >
            10
          </span>
          <span
            class="Text Text--link Text--regular ellipsis--noWrap Tab-selected"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 1
          </span>
          <i
            class="material-symbols material-symbols-rounded Icon Icon--info DismissibleTab-icon--right DismissibleTab-icon--selected cursor-pointer Tab-selected"
            data-test="DesignSystem-DismissibleTabs--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
            tabindex="0"
          >
            clear
          </i>
        </span>
      </div>
      <div
        class="Tab align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="0"
      >
        <span
          class="Tab--regular Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <i
            class="material-symbols material-symbols-rounded Icon Icon--subtle mr-4 "
            data-test="DesignSystem-Tabs--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            call_received
          </i>
          <span
            class="Text Text--subtle Text--regular ellipsis--noWrap"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 2
          </span>
        </span>
      </div>
      <div
        class="Tab Tab--disabled align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="-1"
      >
        <span
          class="Tab--regular Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <span
            class="Pills Badge--secondary Tab-pills Tab-pills--disabled"
            data-test="DesignSystem-Tabs--Pills"
          >
            10
          </span>
          <span
            class="Text Text--disabled Text--regular ellipsis--noWrap"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 3
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabs component 
  activeIndex: 0, onTabChange: "[Function]", withSeparator: false, size: "small"
 1`] = `
<DocumentFragment>
  <div
    class="TabsWrapper"
    data-test="DesignSystem-Tabs"
  >
    <div
      class="TabsWrapper-header"
      data-test="DesignSystem-Tabs--Header"
    >
      <div
        class="Tab Tab--active Tab-selected align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="0"
      >
        <span
          class="Tab--small Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <span
            class="Pills Badge--primary Tab-pills"
            data-test="DesignSystem-Tabs--Pills"
          >
            10
          </span>
          <span
            class="Text Text--link Text--regular ellipsis--noWrap Tab-selected"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 1
          </span>
          <i
            class="material-symbols material-symbols-rounded Icon Icon--info DismissibleTab-icon--right DismissibleTab-icon--selected cursor-pointer Tab-selected"
            data-test="DesignSystem-DismissibleTabs--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
            tabindex="0"
          >
            clear
          </i>
        </span>
      </div>
      <div
        class="Tab align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="0"
      >
        <span
          class="Tab--small Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <i
            class="material-symbols material-symbols-rounded Icon Icon--subtle mr-4 "
            data-test="DesignSystem-Tabs--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            call_received
          </i>
          <span
            class="Text Text--subtle Text--regular ellipsis--noWrap"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 2
          </span>
        </span>
      </div>
      <div
        class="Tab Tab--disabled align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="-1"
      >
        <span
          class="Tab--small Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <span
            class="Pills Badge--secondary Tab-pills Tab-pills--disabled"
            data-test="DesignSystem-Tabs--Pills"
          >
            10
          </span>
          <span
            class="Text Text--disabled Text--regular ellipsis--noWrap"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 3
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabs component 
  activeIndex: 0, onTabChange: "[Function]", withSeparator: true, size: "regular"
 1`] = `
<DocumentFragment>
  <div
    class="TabsWrapper"
    data-test="DesignSystem-Tabs"
  >
    <div
      class="TabsWrapper-header TabsWrapper-header--withSeparator"
      data-test="DesignSystem-Tabs--Header"
    >
      <div
        class="Tab Tab--active Tab-selected align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="0"
      >
        <span
          class="Tab--regular Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <span
            class="Pills Badge--primary Tab-pills"
            data-test="DesignSystem-Tabs--Pills"
          >
            10
          </span>
          <span
            class="Text Text--link Text--regular ellipsis--noWrap Tab-selected"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 1
          </span>
          <i
            class="material-symbols material-symbols-rounded Icon Icon--info DismissibleTab-icon--right DismissibleTab-icon--selected cursor-pointer Tab-selected"
            data-test="DesignSystem-DismissibleTabs--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
            tabindex="0"
          >
            clear
          </i>
        </span>
      </div>
      <div
        class="Tab align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="0"
      >
        <span
          class="Tab--regular Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <i
            class="material-symbols material-symbols-rounded Icon Icon--subtle mr-4 "
            data-test="DesignSystem-Tabs--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            call_received
          </i>
          <span
            class="Text Text--subtle Text--regular ellipsis--noWrap"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 2
          </span>
        </span>
      </div>
      <div
        class="Tab Tab--disabled align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="-1"
      >
        <span
          class="Tab--regular Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <span
            class="Pills Badge--secondary Tab-pills Tab-pills--disabled"
            data-test="DesignSystem-Tabs--Pills"
          >
            10
          </span>
          <span
            class="Text Text--disabled Text--regular ellipsis--noWrap"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 3
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabs component 
  activeIndex: 0, onTabChange: "[Function]", withSeparator: true, size: "small"
 1`] = `
<DocumentFragment>
  <div
    class="TabsWrapper"
    data-test="DesignSystem-Tabs"
  >
    <div
      class="TabsWrapper-header TabsWrapper-header--withSeparator"
      data-test="DesignSystem-Tabs--Header"
    >
      <div
        class="Tab Tab--active Tab-selected align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="0"
      >
        <span
          class="Tab--small Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <span
            class="Pills Badge--primary Tab-pills"
            data-test="DesignSystem-Tabs--Pills"
          >
            10
          </span>
          <span
            class="Text Text--link Text--regular ellipsis--noWrap Tab-selected"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 1
          </span>
          <i
            class="material-symbols material-symbols-rounded Icon Icon--info DismissibleTab-icon--right DismissibleTab-icon--selected cursor-pointer Tab-selected"
            data-test="DesignSystem-DismissibleTabs--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
            tabindex="0"
          >
            clear
          </i>
        </span>
      </div>
      <div
        class="Tab align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="0"
      >
        <span
          class="Tab--small Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <i
            class="material-symbols material-symbols-rounded Icon Icon--subtle mr-4 "
            data-test="DesignSystem-Tabs--Icon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            call_received
          </i>
          <span
            class="Text Text--subtle Text--regular ellipsis--noWrap"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 2
          </span>
        </span>
      </div>
      <div
        class="Tab Tab--disabled align-items-center"
        data-test="DesignSystem-Tabs--Tab"
        tabindex="-1"
      >
        <span
          class="Tab--small Tab--overflow"
          data-test="DesignSystem-Tabs--TextWrapper"
          style="max-width: var(--spacing-9);"
        >
          <span
            class="Pills Badge--secondary Tab-pills Tab-pills--disabled"
            data-test="DesignSystem-Tabs--Pills"
          >
            10
          </span>
          <span
            class="Text Text--disabled Text--regular ellipsis--noWrap"
            data-test="DesignSystem-Tabs--Text"
          >
            Tab 3
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
